scissors/src/main/java/com/lyft/android/scissors/CropView.java

```diff
@@ -97,12 +91,59 @@ private void drawBitmap(Canvas canvas) {
         canvas.drawBitmap(bitmap, transform, bitmapPaint);
     }
 
+    private void drawOverlay(Canvas canvas) {
+        final int viewportWidth = touchManager.getViewportWidth();
+        final int viewportHeight = touchManager.getViewportHeight();
+        final int left = (getWidth() - viewportWidth) / 2;
+        final int top = (getHeight() - viewportHeight) / 2;
+
+        canvas.drawRect(0, top, left, getHeight() - top, viewportPaint);
+        canvas.drawRect(0, 0, getWidth(), top, viewportPaint);
+        canvas.drawRect(getWidth() - left, top, getWidth(), getHeight() - top, viewportPaint);
+        canvas.drawRect(0, getHeight() - top, getWidth(), getHeight(), viewportPaint);
+    }
+
     @Override
     protected void onSizeChanged(int w, int h, int oldw, int oldh) {
         super.onSizeChanged(w, h, oldw, oldh);
         resetTouchManager();
     }
 
+    /**
+     * Returns the aspect ratio of the crop rect and overlay.
+     *
+     * @return The current aspect ratio.
+     */
+    public float getAspectRatio() {
+        return touchManager.getAspectRatio();
+    }
+
+    /**
+     * Returns the effective aspect ratio of the crop rect and overlay, which
+     * equals the native aspect ratio of the image if the set aspect ratio is 0.
+     *
+     * @return The effective aspect ratio.
+     */
+    public float getEffectiveAspectRatio() {
+        float aspect = touchManager.getAspectRatio();
+        if (Float.compare(aspect, 0f) == 0) {
+            aspect = (float) touchManager.getViewportWidth() / touchManager.getViewportHeight();
+        }
+        return aspect;
+    }
```

Comment:
@rharter yeah it's weird when animating from last ratio to `0` we obtain the bitmap and compute target ratio on the activity, that's why I fee like `getAspectRatio` should always return actual ratio in use, see https://github.com/lyft/scissors/pull/27#issuecomment-185031194